src/targets/registry.ts

```diff
@@ -408,20 +445,76 @@ export class RegistryTarget extends BaseTarget {
         this.logger.warn(
           `No files found that match "${onlyIfPresentPattern.toString()}", skipping the target.`
         );
+        this.published = true;
         return undefined;
       }
     }
 
-    const remote = this.registryConfig.registryRemote;
-    const username = await getAuthUsername(this.github);
-    remote.setAuth(username, getGithubApiToken());
-
-    await withTempDir(
-      directory =>
-        this.pushVersionToRegistry(directory, remote, version, revision),
-      true,
-      'craft-release-registry-'
+    const startBatch = RegistryTarget.lock.isLocked();
+    const result = RegistryTarget.lock.runExclusive(() =>
+      this.doPublish(version, revision)
     );
-    this.logger.info('Release registry updated');
+    if (startBatch) {
```

Comment:
Verified this logic with https://gist.github.com/BYK/b7f26582d0a2674f6785b5c80c38de43

Reply:
But in the gist I see `!lock.isLocked()` and here I see `lock.isLocked()` ... the opposite.

Reply:
Since we have test code can we bring it into CI here?

Reply:
Yup you caught a bug! I'll try to add tests here.